blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/EventElasticSearchIO.java

```diff
@@ -0,0 +1,105 @@
+package com.rackspacecloud.blueflood.io;
+
+import com.rackspacecloud.blueflood.service.ElasticClientManager;
+import com.rackspacecloud.blueflood.service.RemoteElasticSearchServer;
+import com.rackspacecloud.blueflood.types.Event;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.search.SearchHit;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.index.query.QueryBuilders.rangeQuery;
+import static org.elasticsearch.index.query.QueryBuilders.termQuery;
+
+public class EventElasticSearchIO implements GenericElasticSearchIO {
+    public static final String EVENT_INDEX = "events";
+    public static final String ES_TYPE = "graphite_event";
+    private final Client client;
+
+    public EventElasticSearchIO() {
+        this(RemoteElasticSearchServer.getInstance());
+    }
+    public EventElasticSearchIO(Client client) {
+        this.client = client;
+    }
+    public EventElasticSearchIO(ElasticClientManager manager) {
+        this(manager.getClient());
+    }
+
+    @Override
+    public void insert(String tenant, List<Map<String, Object>> events) throws Exception {
+        for (Map<String, Object> event : events) {
+            event.put(Event.FieldLabels.tenantId.toString(), tenant);
```

Comment:
We should batch requests the way we do it for writing metrics index instead of writing each request individually.